### PR TITLE
inform service about getthread calls as result of foregrounding the app

### PR DESCRIPTION
**Pull request by mmaxim**
Otherwise we can get gray bars when foregrounding into a thread.

### CODE REVIEW

File: shared/actions/chat2/index.js [hunk 1]
```diff
@@ -845,6 +845,16 @@ const loadMoreMessages = (
       ''} num: ${numberOfMessagesToLoad} reason: ${reason}`
   )
 
+  const rpcReason = (function(r) {
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: Just use let and not do an inline function call

Comment (mmaxim) [reply shared/actions/chat2/index.js]: How come? This seemed shorter.

Comment (chrisnojima) [reply shared/actions/chat2/index.js]: Then hoist it up and call it

Comment (mmaxim) [reply shared/actions/chat2/index.js]: Is it just a style thing, or is this dangerous? I'll move it, just curious.

Comment (chrisnojima) [reply shared/actions/chat2/index.js]: It's a style thing. We never use iife.

**Pull request merged**